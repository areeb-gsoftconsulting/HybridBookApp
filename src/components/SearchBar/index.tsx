import React from 'react';
import { useStyles } from './styles'
import {
    StyleSheet,
    TextInput,
    View,
    TouchableOpacity
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import { initReactI18next, useTranslation } from "react-i18next";



const SearchBar = () => {
    const styles = useStyles()
    const [t, i18n] = useTranslation()

    return (
      <View style={styles.searchView}>
        <TextInput
          testID="searchInput"
          placeholder={t('SearchBar.search')}
          style={styles.searchViewInput}
        />
        <TouchableOpacity testID={'searchBookies'}>
          <Icon name="search" size={25} style={styles.searchViewChildren} />
        </TouchableOpacity>
      </View>
    );
};



export default SearchBar;
