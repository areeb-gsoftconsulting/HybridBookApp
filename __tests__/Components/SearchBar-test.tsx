import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import SearchBar from '../../src/components/SearchBar';

describe('Component testing',()=>{
    test('Snapshot', () => {
      const tree = render(<SearchBar />);
      expect(tree).toMatchSnapshot();
    });

    test('search action', () => {
      const {getByTestId} = render(<SearchBar />);
      fireEvent.press(getByTestId('searchBookies'));
    });
    test('search input change', () => {
      const {getByTestId} = render(<SearchBar />);
      // fireEvent.press(getByTestId('search'));
      fireEvent.changeText(getByTestId('searchInput'));
    });
});
