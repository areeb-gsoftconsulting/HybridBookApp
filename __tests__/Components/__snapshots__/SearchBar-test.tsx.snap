// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component testing Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": undefined,
      "borderColor": "grey",
      "borderRadius": 50,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 20,
      "padding": 5,
      "width": "90%",
    }
  }
>
  <TextInput
    placeholder="SearchBar.search"
    style={
      Object {
        "color": "#6200ee",
        "margin": 2,
        "marginLeft": 5,
        "width": undefined,
      }
    }
    testID="searchInput"
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="searchBookies"
  >
    <mockedDefaultExport
      name="search"
      size={25}
      style={
        Object {
          "color": "#6200ee",
          "marginRight": 15,
        }
      }
    />
  </View>
</View>
`;
